sonar-core/src/main/java/org/sonar/core/event/EventDto.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.core.event;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+public class EventDto {
+
+  public static final String CATEGORY_VERSION = "Version";
+  public static final String CATEGORY_ALERT = "Alert";
+  public static final String CATEGORY_PROFILE = "Profile";
+
+  private Long id;
+
+  private String name;
+
+  private String description;
+
+  private String category;
+
+  private Long date;
+
+  private Long createdAt;
+
+  private String data;
+
+  private Long snapshotId;
+
+  private String componentUuid;
+
+  public Long getId() {
+    return id;
+  }
+
+  public EventDto setId(Long id) {
+    this.id = id;
+    return this;
+  }
+
+  public String getCategory() {
+    return category;
+  }
+
+  public EventDto setCategory(String category) {
+    this.category = category;
+    return this;
+  }
+
+  public String getComponentUuid() {
+    return componentUuid;
+  }
+
+  public EventDto setComponentUuid(String componentUuid) {
+    this.componentUuid = componentUuid;
+    return this;
+  }
+
+  public Long getCreatedAt() {
+    return createdAt;
+  }
+
+  public EventDto setCreatedAt(Long createdAt) {
+    this.createdAt = createdAt;
+    return this;
+  }
+
+  @CheckForNull
+  public String getData() {
+    return data;
+  }
+
+  public EventDto setData(@Nullable String data) {
+    this.data = data;
+    return this;
+  }
+
+  public Long getDate() {
```

Comment:
I prefer long, and add a helper that return a Date if needed. As we're now using long for every dates that we're using, using long everywhere seems good to me.
The only place to use a Date is in WS and in E/S.